### PR TITLE
RSS-K8: Adding the server sequence service registry to shuffle manager

### PR DESCRIPTION
**Pull request by mabansal**
We have added the new server sequence service registry into the metadata 
In this pull request, we are adding the server sequence service registry to the shuffle manager  
This will let the client use the new registry service in the K8 environment

### CODE REVIEW

File: src/main/scala/org/apache/spark/shuffle/RssOpts.scala [hunk 1]
```diff
@@ -130,11 +135,32 @@ object RssOpts {
...
+      .stringConf
+      .createWithDefault("rss-%s.rss.remote-shuffle-service.svc.cluster.local:9338")
+  val serverSequenceStartIndex: ConfigEntry[Int] =
+    ConfigBuilder("spark.shuffle.rss.serverSequence.startIndex")
+      .doc("Server sequence start index")
+      .intConf
+      .createWithDefault(0)
+  val serverSequenceEndIndex: ConfigEntry[Int] =
+    ConfigBuilder("spark.shuffle.rss.serverSequence.endIndex")
+      .doc("Server sequence start index")
```
Comment (vectorijk) [inline src/main/scala/org/apache/spark/shuffle/RssOpts.scala:155]: typo : )
```suggestion
      .doc("Server sequence end index")
```

**Review by vectorijk — APPROVED** [review]

**Comment by mayurdb** [comment]
We will have to implement the registerServer call in the this service registry and also make the clients connect to this service registry. Current code would throw an exception

Reference: 
https://github.com/uber/RemoteShuffleService/commit/9daa994e149354282315ee517630491d53c2bb72#diff-4824c8eccd34d55acbc91b62165f7b17680267ed61aab0b90dc05b3c9fdf90baR60

https://github.com/uber/RemoteShuffleService/blob/master/src/main/scala/org/apache/spark/shuffle/RssServiceRegistry.scala#L54

**Comment by mabansal** [comment]
> We will have to implement the registerServer call in the this service registry and also make the clients connect to this service registry. Current code would throw an exception
> 
> Reference:
> [9daa994#diff-4824c8eccd34d55acbc91b62165f7b17680267ed61aab0b90dc05b3c9fdf90baR60](https://github.com/uber/RemoteShuffleService/commit/9daa994e149354282315ee517630491d53c2bb72#diff-4824c8eccd34d55acbc91b62165f7b17680267ed61aab0b90dc05b3c9fdf90baR60)
> 
> https://github.com/uber/RemoteShuffleService/blob/master/src/main/scala/org/apache/spark/shuffle/RssServiceRegistry.scala#L54

@mayurdb As of now, the implementation is very easy for K8, where we will have a set of servers that will be given by K8 to RSS and K8 will make sure those servers are up and running. However, you are right we need to think of other solutions than ZK for K8 for getting more dynamic servers. Will add that feature later.

**Comment by mayurdb** [comment]
lgtm. There are more changes planned in follow up diffs, which will address these comments.

**Comment by mayurdb** [comment]
We have few follow ups -
1. Fixing client code to connect to service registry or fallback when there is no valid registry
2. Adding service registry for K8s
3. CI integration for running UTs for k8s (and actually master branch as well)

**Comment by mabansal** [comment]
> We have few follow ups -
> 
> 1. Fixing client code to connect to service registry or fallback when there is no valid registry
> 2. Adding service registry for K8s
> 3. CI integration for running UTs for k8s (and actually master branch as well)

1. For client code we don't need a service registry, currently, it uses config to discover server however we will implement this once we have 2 implemented
2. We have to do this, it's a longer-term, will work on it as a separate project
3. I will do that in followup JIRA

File: src/main/scala/org/apache/spark/shuffle/RssOpts.scala (hunk 1 above)
Comment (mabansal) [reply src/main/scala/org/apache/spark/shuffle/RssOpts.scala:155]: > typo : )

Done

**Review by mayurdb — APPROVED** [review]

**Review by vectorijk — APPROVED** [review]
lgtm

**Comment by mabansal** [comment]
Thanks , merging the PR

**Pull request merged**